src/senaite/core/content/person.py

```diff
@@ -0,0 +1,578 @@
+        """Return possible address types
+        """
+        return ["PhysicalAddress", "PostalAddress"]
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setFullname(self, value):
+        parts = value.split(" ")
+        length = len(parts)
+        if length == 2:
+            self.setFirstname(parts[0])
+            self.setSurname(parts[1])
+        elif length == 3:
+            self.setFirstname(parts[0])
+            self.setMiddlename(parts[1])
+            self.setSurname(parts[2])
+        elif length == 4:
+            self.setFirstname(parts[0])
+            self.setMiddleinitial(parts[1])
+            self.setMiddlename(parts[2])
+            self.setSurname(parts[3])
+        else:
+            self.setSurname(value)
+
+    def getFullname(self):
+        """Person's Fullname
+        """
+        fn = self.getFirstname()
+        mi = self.getMiddleinitial()
+        md = self.getMiddlename()
+        sn = self.getSurname()
+        fullname = ""
+        if fn or sn:
+            if mi and md:
+                fullname = "%s %s %s %s" % (
+                    self.getFirstname(),
+                    self.getMiddleinitial(),
+                    self.getMiddlename(),
+                    self.getSurname())
+            elif mi:
+                fullname = "%s %s %s" % (
+                    self.getFirstname(),
+                    self.getMiddleinitial(),
+                    self.getSurname())
+            elif md:
+                fullname = "%s %s %s" % (
+                    self.getFirstname(),
+                    self.getMiddlename(),
+                    self.getSurname())
+            else:
+                fullname = "%s %s" % (self.getFirstname(), self.getSurname())
+        return fullname.strip()
+
+    @security.protected(permissions.View)
+    def getSalutation(self):
+        accessor = self.accessor("salutation")
+        value = accessor(self) or ""
+        return safe_unicode(value).encode("utf-8")
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setSalutation(self, value):
+        mutator = self.mutator("salutation")
+        mutator(self, safe_unicode(value))
+
+    # BBB: AT schema field property
+    Salutation = property(getSalutation, setSalutation)
+
+    @security.protected(permissions.View)
+    def getFirstname(self):
+        accessor = self.accessor("firstname")
+        value = accessor(self) or ""
+        return safe_unicode(value).encode("utf-8")
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setFirstname(self, value):
+        mutator = self.mutator("firstname")
+        mutator(self, safe_unicode(value))
+
+    # BBB: AT schema field property
+    Firstname = property(getFirstname, setFirstname)
+
+    @security.protected(permissions.View)
+    def getMiddleinitial(self):
+        accessor = self.accessor("middleinitial")
+        value = accessor(self) or ""
+        return safe_unicode(value).encode("utf-8")
```

Comment:
If you use `senaite.core.schema.TextLineField` instead of `zope.schema.TextLine`, the field's accessor and mutator automatically handle the necessary encoding. See: [https://github.com/senaite/senaite.core/blob/2.x/src/senaite/core/schema/textlinefield.py](https://github.com/senaite/senaite.core/blob/2.x/src/senaite/core/schema/textlinefield.py)